internal/types/compare.go

```diff
@@ -255,3 +251,121 @@ func compareNumbers(a float64, b int64) CompareResult {
 
 	return CompareResult(bigA.Cmp(bigB))
 }
+
+// dataTypeOrderResult represents the comparison order of data types.
+type dataTypeOrderResult uint8
+
+const (
+	_ dataTypeOrderResult = iota
+	minKeyDataType
+	nullDataType
+	nanDataType
+	numbersDataType
+	stringDataType
+	objectDataType
+	arrayDataType
+	binDataType
+	objectIdDataType
+	booleanDataType
+	dateDataType
+	timestampDataType
+	regexDataType
+	maxKeyDataType
+)
+
+// defineDataType define which type has value and returns a sequence the type has.
+func defineDataType(value any) dataTypeOrderResult {
+	switch value := value.(type) {
+	case NullType:
```

Comment:
Please rearrange cases in order with [code style](https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md#code-style-and-conventions)

Reply:
fixed.